### PR TITLE
Feat/fetch data

### PR DESCRIPTION
**Pull request by nattsnb**

### CODE REVIEW

File: DeleteButton.js [hunk 1]
```diff
@@ -0,0 +1,19 @@
+export class DeleteButton {
+    constructor(iD, userContainer) {
+        this.url = 'https://jsonplaceholder.typicode.com/users'
+        this.iD = iD + 1;
```
Comment (Solnick) [inline DeleteButton.js]:
```suggestion
        this.id = id + 1;
```

File: DeleteButton.js [hunk 2]
```diff
@@ -0,0 +1,19 @@
...
+        console.log(this.iD)
+        const deleteButton = document.createElement("button");
+        deleteButton.classList.add('delete-button');
+        deleteButton.innerHTML = "DELETE"
+        userContainer.append(deleteButton);
+        deleteButton.addEventListener("click", this.deleteUser)
```
Comment (Solnick) [inline DeleteButton.js]: create a separate funciton for the button initialization

File: DeleteButton.js [hunk 3]
```diff
@@ -0,0 +1,19 @@
...
+    }
+
+    deleteUser = () => {
+        const deleteUser =  fetch(this.url + '/' + this.iD, {
```
Comment (Solnick) [inline DeleteButton.js]: use string literal instead of string concatenation

Comment (Solnick) [reply DeleteButton.js]: you are assigning the promise returned from fetch to the `const deleteUser`, but the promise is never awaited, and you have no control over if the backend returned an Error or a Sucess.

File: DeleteButton.js [hunk 4]
```diff
@@ -0,0 +1,19 @@
+export class DeleteButton {
+    constructor(iD, userContainer) {
+        this.url = 'https://jsonplaceholder.typicode.com/users'
+        this.iD = iD + 1;
+        console.log(this.iD)
```
Comment (Solnick) [inline DeleteButton.js]: get rid of the console.logs. Remeber to NEVER put console.log in the PR-ready code

File: style.css [hunk 5]
```diff
@@ -1,5 +1,16 @@
...
+
+.user-container {
+  border-style: solid;
+  border-width: 1px;
+  display: flex;
+  flex-direction: row;
+  justify-content: space-between;
+}
+
+.delete-button {
```
Comment (Solnick) [inline style.css]: delete it

**Review by Solnick — CHANGES_REQUESTED** [review]

File: DeleteButton.js (hunk 1 above)
Comment (nattsnb) [reply DeleteButton.js]: done

File: DeleteButton.js (hunk 2 above)
Comment (nattsnb) [reply DeleteButton.js]: done

File: DeleteButton.js (hunk 3 above)
Comment (nattsnb) [reply DeleteButton.js]: 1st done

File: DeleteButton.js (hunk 4 above)
Comment (nattsnb) [reply DeleteButton.js]: done

File: DeleteButton.js (hunk 3 above)
Comment (nattsnb) [reply DeleteButton.js]: 2nd done but im not sure

File: style.css (hunk 5 above)
Comment (nattsnb) [reply style.css]: done

File: DeleteButton.js [hunk 6]
```diff
@@ -0,0 +1,25 @@
...
+  }
+
+  deleteUser = async () => {
+    const deleteUserResponse = await fetch(`${this.url}/${this.userId}`, {
+      method: "DELETE",
+    });
+    if (this.userContainer) {
+      this.userContainer.remove();
+    }
+  };
```
Comment (Solnick) [inline DeleteButton.js]: `deleteUserResponse ` is unused.

You need to check if the request was successful before removing the user tile

File: UserList.js [hunk 7]
```diff
@@ -1,46 +1,41 @@
...
+export class UserList {
+  constructor() {
+    this.fetchedData = {};
+    this.startButton = null;
+    this.fetchData();
+    this.initializeStartButton();
+  }
+  async fetchData() {
+    const response = await fetch("https://jsonplaceholder.typicode.com/users");
+    this.fetchedData = await response.json();
```
Comment (Solnick) [inline UserList.js]:
```suggestion
    this.users = await response.json();
```

File: UserList.js [hunk 8]
```diff
@@ -1,46 +1,41 @@
...
+  };
 
+  initializeList = () => {
+    const listContainer = document.querySelector("#app");
+    for (let i = 0; i < this.fetchedData.length; i++) {
+      const userContainer = document.createElement("div");
+      listContainer.append(userContainer);
+      userContainer.classList.add("user-container");
+      userContainer.innerHTML = this.fetchedData[i].name;
+      const deleteButton = new DeleteButton(
```
Comment (Solnick) [inline UserList.js]: the `deleteButton` variable is unused

File: counter.js [hunk 9]
```diff
@@ -1,9 +1,9 @@
 export function setupCounter(element) {
-  let counter = 0
+  let counter = 0;
```
Comment (Solnick) [inline counter.js]: you don't need this file

**Review by Solnick — CHANGES_REQUESTED** [review]

File: DeleteButton.js [hunk 10]
```diff
@@ -10,7 +10,7 @@ export class DeleteButton {
     const deleteUserResponse = await fetch(`${this.url}/${this.userId}`, {
       method: "DELETE",
     });
-    if (this.userContainer) {
+    if (this.userContainer && deleteUserResponse) {
```
Comment (Solnick) [inline DeleteButton.js]: `deleteUserResponse` will always be truthy. You are not checking the actual response status, please follow the documentation to find solution https://developer.mozilla.org/en-US/docs/Web/API/Response/ok

**Review by Solnick — CHANGES_REQUESTED** [review]

**Review by Solnick — APPROVED** [review]

**Pull request merged**